Userland/Services/WindowServer/Window.cpp

```diff
@@ -507,33 +508,34 @@ void Window::handle_keydown_event(KeyEvent const& event)
     if (event.modifiers() == Mod_Alt && event.code_point() && m_menubar.has_menus()) {
         // When handling alt shortcuts, we only care about the key that has been pressed in addition
         // to alt, not the code point that has been mapped to alt+[key], so we have to look up the
-        // scancode directly from the "unmodified" character map.
+        // kernel map_entry_index value directly from the "unmodified" character map.
         auto character_map_or_error = Keyboard::CharacterMap::fetch_system_map();
```

Comment:
Ok, I'm confused about something here.

This "mode" of the PS2 Keyboard. It's static right? After init anyway. We figure out at boot time what "keymap set" is going to be correct for the keyboard device, and then it can't change. We can't hotplug PS2. You can't plug PS2 in after boot. Are any of these assumptions incorrect?

If I have the correct understanding, then why can't the kernel just give us a system character map that represents the actual attached hardware? I don't understand why we need to pass this extra information to userspace in each key event when we could simply present the properly indexed map data to the user via this "fetch system map" call here

Reply:
What about USB HID then? it's not really useful to hardcode things just to make them flexible later on, especially when USB HID is just around the corner.

Reply:
What I'm trying to suggest is that we should know, at the time that the user fetches the system character map, which keyboard is active, and what keymap it has. And return that keymap pointer in the `getkeymap` syscall. So I don't see a reason to add this extra complexity to the userspace code when the kernel has all the information.